src/effects/VST3/VST3Effect.cpp

```diff
@@ -1233,3 +1177,54 @@ void VST3Effect::ReloadUserOptions()
 
    SetBlockSize(mUserBlockSize);
 }
+
+
+bool VST3Effect::TransferDataToWindow(const EffectSettings& settings)
+{
+   const auto vst3Settings = GetSettings(settings);
+
+   const bool scanOK = ForEachParameter(
+
+      [&](const ParameterInfo& pi)
+      {
+         if (mComponentHandler == nullptr)
+            return false;
+
+         const auto id = pi.id;
+
+         auto itr = vst3Settings.mValues.find(id);
+         if (itr != vst3Settings.mValues.end())
+         {
+            const auto& value = itr->second;
+
+            // set the value to the handler at the id
+            if (mComponentHandler->beginEdit(id) == Steinberg::kResultOk)
+            {
+               auto cleanup = finally([&] {
+                  mComponentHandler->endEdit(id);
+               });
+               mComponentHandler->performEdit(id, value);
+            }
+            else
+            {
+               return false;
+            }
+         }
+         else
+         {
+            return false;
+         }
+
+         return true;
+      }
+
+   );
+
+   EffectSettings dummySettings;
+   SyncParameters(dummySettings);
```

Comment:
This on the other hand is here because a call to `SyncParameters` is deleted from `LoadUserPreset`.  But it uses a dummy settings, which is strange; but that is because `SyncParameters` takes a non-const reference to settings, as it must in other contexts.

So I think the visit above should be deleted, and these two lines should be retained, but some comment added to explain that this is only transitional, pending the definition of the Instance class.

We need to discuss with @vsverchinsky .